src/api/Annotations.js

```diff
@@ -70,24 +78,49 @@ export default class Annotations extends MarkerBasedApi {
         });
     }
 
-    updateAnnotation(
-        fileId: string,
+    updateAnnotation({
+        fileId,
+        annotationId,
+        permissions,
+        message,
+        status,
+        successCallback,
+        errorCallback,
+    }: {
         annotationId: string,
+        errorCallback: (e: ElementsXhrError, code: string) => void,
+        fileId: string,
+        message?: string,
         permissions: AnnotationPermission,
-        message: string,
+        status?: AnnotationStatus,
         successCallback: (annotation: Annotation) => void,
-        errorCallback: (e: ElementsXhrError, code: string) => void,
-    ): void {
+    }): void {
         this.errorCode = ERROR_CODE_EDIT_ANNOTATION;
 
-        try {
-            this.checkApiCallValidity(PERMISSION_CAN_EDIT, permissions, fileId);
-        } catch (e) {
-            errorCallback(e, this.errorCode);
-            return;
+        if (message) {
+            try {
+                this.checkApiCallValidity(PERMISSION_CAN_EDIT, permissions, fileId);
+            } catch (e) {
+                errorCallback(e, this.errorCode);
+                return;
+            }
         }
 
-        const requestData = { data: { description: { message } } };
+        if (status) {
+            try {
+                this.checkApiCallValidity(PERMISSION_CAN_RESOLVE, permissions, fileId);
+            } catch (e) {
+                errorCallback(e, this.errorCode);
+                return;
+            }
+        }
+
+        const requestData = {
+            data: {
+                ...(message ? { description: { message } } : null),
```

Comment:
Minor, but I believe Axios removes properties with a value of `undefined`. Can we simplify to the below:

```
data: {
    description: message,
    status,
}